tests/helpers_tests/test_condition.py

```diff
@@ -290,19 +290,27 @@ def test_exact__given_number_values__calls_is_finite_number(self):
         exact_int_condition_list, {'lasers_count': 9000}
       )
 
+    # assert that isFiniteNumber only needs to reject condition value to stop evaluation.
     with mock.patch('optimizely.helpers.validator.is_finite_number',
-                    return_value=True) as mock_is_finite:
-      self.assertTrue(evaluator.evaluate(0))
-    mock_is_finite.assert_called_with(9000)
+                    side_effect=[False, True]) as mock_is_finite:
```

Comment:
@nchilada In exact match need to send exactly same values to ensure that it's the mocked is_finite that is really driving the result. Therefore, can't make a value-base side-effect method as in gt/lt methods. 
This returns False on 1st call and True on 2nd.

Reply:
Ah, you're right. Good solution. Thanks @oakbani!